lib/json_api_client/middleware/status.rb

```diff
@@ -11,8 +21,11 @@ def call(environment)
             handle_status(code, env)
           end
         end
-      rescue Faraday::ConnectionFailed, Faraday::TimeoutError
-        raise Errors::ConnectionError, environment
+      rescue => ex
+        if CONNECTION_ERRORS.include?(ex.class)
```

Comment:
Instead of checking against hardcoded list of strings, wouldn't it be better to check whether `ex.class` is a valid class, reraising if so, and raising an `Errors::ConnectionError` if not?